tests/unit/azure/test_azure_credentials.py

```diff
@@ -0,0 +1,428 @@
+    prompts = MockPrompts({"Have you reviewed the azure_storage_account_info.csv *": "Yes"})
+
+    assert isinstance(AzureServicePrincipalMigration.for_cli(ws, prompts), AzureServicePrincipalMigration)
+
+
+def test_list_storage_credentials(ws):
+    ws.storage_credentials.list.return_value = [
+        StorageCredentialInfo(aws_iam_role=AwsIamRole(role_arn="arn:aws:iam::123456789012:role/example-role-name")),
+        StorageCredentialInfo(
+            azure_managed_identity=AzureManagedIdentity(
+                access_connector_id="/subscriptions/.../providers/Microsoft.Databricks/..."
+            )
+        ),
+        StorageCredentialInfo(
+            azure_service_principal=AzureServicePrincipal(
+                application_id="b6420590-5e1c-4426-8950-a94cbe9b6115",
+                directory_id="62e43d7d-df53-4c64-86ed-c2c1a3ac60c3",
+                client_secret="secret",
+            )
+        ),
+    ]
+
+    sp_migration = AzureServicePrincipalMigration(MagicMock(), ws, MagicMock(), MagicMock())
+
+    expected = {"b6420590-5e1c-4426-8950-a94cbe9b6115"}
+    sp_migration._list_storage_credentials()
+
+    assert expected == sp_migration._list_storage_credentials()
+
+
+def test_list_storage_credentials_for_integration_test(ws):
+    ws.storage_credentials.list.return_value = [
+        StorageCredentialInfo(aws_iam_role=AwsIamRole(role_arn="arn:aws:iam::123456789012:role/example-role-name")),
+        StorageCredentialInfo(
+            azure_managed_identity=AzureManagedIdentity(
+                access_connector_id="/subscriptions/.../providers/Microsoft.Databricks/..."
+            )
+        ),
+        StorageCredentialInfo(
+            name="spn_for_integration_test",
+            azure_service_principal=AzureServicePrincipal(
+                application_id="b6420590-5e1c-4426-8950-a94cbe9b6115",
+                directory_id="62e43d7d-df53-4c64-86ed-c2c1a3ac60c3",
+                client_secret="secret",
+            ),
+        ),
+    ]
+
+    # test storage credential: spn_for_integration_test is picked up
+    # in integration test, we only pick up the existing storage credential created in integration test and ignore the others
+    sp_migration = AzureServicePrincipalMigration(
+        MagicMock(), ws, MagicMock(), MagicMock(), integration_test_flag="spn_for_integration_test"
+    )
+    expected = {"b6420590-5e1c-4426-8950-a94cbe9b6115"}
+    sp_migration._list_storage_credentials()
+    assert expected == sp_migration._list_storage_credentials()
+
+    # test storage credential is not picked up
+    # if integration test does not create storage credential, we use dummy integration_test_flag to filter out other existing storage credentials
+    sp_migration = AzureServicePrincipalMigration(
+        MagicMock(), ws, MagicMock(), MagicMock(), integration_test_flag="other_spn"
+    )
+    sp_migration._list_storage_credentials()
+    assert {""} == sp_migration._list_storage_credentials()
```

Comment:
it's not allowed to call private methods in unit tests. rewrite unit tests to call only public methods.

Reply:
done